### PR TITLE
Pythranized version of _matfuncs_sqrtm

### PR DESCRIPTION
**Pull request by serge-sans-paille**

### CODE REVIEW

File: scipy/linalg/_matfuncs_sqrtm.py
Comment (tupui) [inline scipy/linalg/_matfuncs_sqrtm.py:90]: I think it should be
```suggestion
        raise SqrtmError(*e.args) from e
```

**Review by tupui — COMMENTED** [review]
Thanks @serge-sans-paille. Do you have some benchmarks? Are there other benefits than the better maintainability due to Pythran here?

**Comment by serge-sans-paille** [comment]
Thanks @tupui for the feedback. I ran the following benchmarks:

```console
$ python -m timeit -s 'from scipy.linalg._matfuncs_sqrtm import _sqrtm_triu; from numpy import tri; x = tri(100)*3' '_sqrtm_triu(x)'
```

then

```console
$ python -m timeit -s 'from scipy.linalg._matfuncs_sqrtm import _sqrtm_triu; from numpy import tri; x = tri(1000)*3' '_sqrtm_triu(x)'
```

And got the following timings

size  | pythran  | cython
------|----------|-------
100   |    221us | 221us
1000  | 113ms    | 117ms

So pythran is slightly faster, but that's neglectable

File: scipy/linalg/_matfuncs_sqrtm.py
Comment (serge-sans-paille) [reply scipy/linalg/_matfuncs_sqrtm.py]: I'm so glad I lerant something today. Thanks @tupui, that syntax was new to me!

**Review by tupui — APPROVED** [review]
LGTM thanks @serge-sans-paille. Let's just wait for @rgommers to not create more merge conflicts on the meson work.

**Review by tylerjereddy — APPROVED** [review]
Alright, there's a core developer approval and CI is passing. I don't personally see a massive benefit to Pythran usage here, but happy to support giving Pythran more in-the-wild usage since some folks are keen on it.

I wonder if Pythran has the ability to turn on linetracing for codecov reporting (code not covered in this PR). Actually, I think we may not have this working reliably for Cython cov either.

**Comment by tylerjereddy** [comment]
thanks!

**Pull request merged**
